File: primes/lua/compute.lua
local PRIME_COUNT = assert(
    os.getenv("PRIME_COUNT"),
    "Please set PRIME_COUNT environment variable."
)
PRIME_COUNT = tonumber(PRIME_COUNT)
local BENCH_DEBUG = os.getenv("BENCH_DEBUG")

local primes = {}
local function is_prime(number)
    for _, prime in ipairs(primes) do
        if number % prime == 0 then
            return false
        end
    end
    return true
end

print("begin")
io.stdout:flush()

local number = 2
while #primes < PRIME_COUNT do
    if is_prime(number) then
        table.insert(primes, number)
    end
    number = number + 1
end

if BENCH_DEBUG == "true" then
    print(table.concat(primes, "\n"))
end

print("end")
io.stdout:flush()

File: primes/nodejs/compute.js
const PRIME_COUNT = Number(process.env.PRIME_COUNT);
if(isNaN(PRIME_COUNT)) {
    console.log("Please set PRIME_COUNT environment variable.");
    process.exit(-1);
}
const BENCH_DEBUG = process.env.BENCH_DEBUG;

const primes = [];
function is_prime(n) {
    for(let i = 0; i < primes.length; i++) {
        if(n % primes[i] == 0) {
            return false;
        }
    }
    return true;
}

let number = 2;
process.stdout.write("begin\n", function() {
    let result = "";
    while(primes.length < PRIME_COUNT) {
        if(is_prime(number)) {
            primes.push(number);
        }
        number++;
    }
    if(BENCH_DEBUG === "true") {
        var primes_str = "";
        for(var i = 0; i < primes.length; i++) {
            primes_str += primes[i]+"\n";
        }
        console.log(primes_str.slice(0, -1));
    }
    process.stdout.write("end\n", function() {
        process.exit(0);
    });
});
